lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/dataset/DatasetAggregator.scala

```diff
@@ -40,6 +40,46 @@ private[lightgbm] object ChunkedArrayUtils {
         chunkedArray.getItem(chunkCount, lastChunkIdx, defaultVal))
     }
   }
+
+  def insertTransposedChunkedArray[T: Numeric](chunkedArray: ChunkedArray[T],
```

Comment:
Note that I tried to put the transpose in several places, and there were some that were more efficient and easier to read, but they would require changes in the LightGBM library to add functionality.  I figured it was better to just fix it with SynapseML changes only since we are moving to streaming, where this is not a problem.